### PR TITLE
feat: adding signin with password

### PR DESCRIPTION
**Pull request by DarrylD**
starting #4

### CODE REVIEW

File: dist/src/index.js [hunk 1]
```diff
@@ -68,6 +68,13 @@ var Firedux = function () {
     if (options.url) {
       console.warn('Firedux option "url" is deprecated, use "ref" instead.');
     }
+
+    this.v3 = options.ref !== null;
```
Comment (adjohnson916) [inline dist/src/index.js]: Should use `Firebase.SDK_VERSION` checking for `3.` prefix.

File: src/index.js [hunk 2]
```diff
@@ -33,6 +33,13 @@ export default class Firedux {
     if (options.url) {
       console.warn('Firedux option "url" is deprecated, use "ref" instead.')
     }
+
+    this.v3 = options.ref !== null
```
Comment (adjohnson916) [inline src/index.js]: Should use `Firebase.SDK_VERSION` checking for `'3.'` prefix. We support `ref` option in both versions.

File: src/index.js [hunk 3]
```diff
@@ -33,6 +33,13 @@ export default class Firedux {
...
+
+    if (this.v3) {
+        this.auth = firebase.auth// eslint-disable-line
```
Comment (adjohnson916) [inline src/index.js]: Use `Firebase` not `firebase` or change the import to lowercase.

File: src/index.js [hunk 4]
```diff
@@ -153,23 +175,36 @@ export default class Firedux {
...
+          // need to ignore redux
+          if (_.isFunction(credentials)) return null
+
+          // TODO add custom later...
+          this.auth()
+            .signInWithEmailAndPassword(credentials.email, credentials.password)
+            .then(authData => handler(null, authData))
+            .catch(error => handleError(error))
+        }
+        else if (credentials.token) { // eslint-disable-line brace-style
```
Comment (adjohnson916) [inline src/index.js]: Don't violate eslint standard here - collapse with preceding brace to single line.

File: src/index.js [hunk 5]
```diff
@@ -183,9 +218,15 @@ export default class Firedux {
   }
   logout () {
     const { dispatch } = this
+
     return new Promise((resolve, reject) => {
       dispatch({type: 'FIREBASE_LOGOUT_ATTEMPT'})
-      this.ref.unauth()
+      if (this.v3) {
+        this.auth().signOut()
```
Comment (adjohnson916) [inline src/index.js]: This returns a promise, we should resolve/reject with it.

File: src/index.js [hunk 6]
```diff
@@ -153,23 +175,36 @@ export default class Firedux {
...
         that.authData = authData
         dispatch({type: 'FIREBASE_LOGIN', authData: authData, error: error})
         resolve(authData)
       }
 
       try {
-        if (credentials.token) {
+        if (this.v3) {
+          // need to ignore redux
+          if (_.isFunction(credentials)) return null
```
Comment (adjohnson916) [inline src/index.js]: Help me understand this line - when is `credentials` a function?

File: src/index.js [hunk 7]
```diff
@@ -132,6 +139,21 @@ export default class Firedux {
         }))
       }
 
+      if (this.v3) {
+        const auth = this.auth()
+
+        auth.onAuthStateChanged(function (user) {
+          if (user) {
+            resolve(user)
```
Comment (adjohnson916) [inline src/index.js]: We should probably persist the `user` here either as `this.authData` or some new property.

File: src/index.js (hunk 3 above)
Comment (DarrylD) [inline src/index.js]: seems to be a version conflict 

`firebase.SDK_VERSION` = "3.2.1"
`Firebase.SDK_VERSION`= "2.4.2"

**Comment by adjohnson916** [comment]
Closing in favor of #89.

**Pull request closed without merging**